frontend/www/js/omegaup/components/arena/FeedbackCodeView.vue

```diff
@@ -0,0 +1,173 @@
+<template>
+  <div>
+    <div class="container-fluid" data-feedback-code-mirror>
+      <textarea v-show="false" ref="cm-editor" v-model="value"></textarea>
+    </div>
+    <div class="container-fluid text-right py-2">
+      <button
+        class="btn btn-primary mx-2"
+        :disabled="!numberOfComments"
+        @click.prevent="saveFeedbackList"
+      >
+        {{ T.submissionSendFeedback }}
+      </button>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop, Ref } from 'vue-property-decorator';
+import T from '../../lang';
+import CodeMirror from 'codemirror';
+import { EditorOptions, languageModeMap, modeList } from './CodeView.vue';
+import Feedback, { ArenaCourseFeedback, FeedbackStatus } from './Feedback.vue';
+
+for (const mode of modeList) {
+  require(`codemirror/mode/${mode}/${mode}.js`);
+}
+
+@Component({
+  components: {},
+})
+export default class FeedbackCodeView extends Vue {
+  @Prop() language!: string;
+  @Prop() value!: string;
+  @Prop({ default: () => new Map() }) feedbackList!: Map<
+    number,
+    ArenaCourseFeedback
+  >;
+  @Ref('cm-editor') private readonly cmEditor!: HTMLTextAreaElement;
+
+  T = T;
+  mode = languageModeMap[this.language] ?? languageModeMap['cpp17-gcc'];
+  numberOfComments = 0;
+
+  get feedbackListEmpty(): number {
+    return this.feedbackList.size;
+  }
+
+  get editorOptions(): EditorOptions {
+    return {
+      tabSize: 2,
+      lineNumbers: true,
+      mode: this.mode,
+      readOnly: true,
+      gutters: ['CodeMirror-linenumbers', 'breakpoints'],
+    };
+  }
+
+  recalculateNumberOfComments(num: number): number {
+    return num;
+  }
+
+  mounted() {
+    const editor = CodeMirror.fromTextArea(this.cmEditor, this.editorOptions);
+    let lineWidget: null | CodeMirror.LineWidget = null;
+
+    editor.on(
+      'gutterClick',
+      (codeMirror: CodeMirror.Editor, numberOfLine: number) => {
+        if (this.feedbackList.get(numberOfLine)) {
+          return;
+        }
+        lineWidget = codeMirror.addLineWidget(
```

Comment:
```suggestion
        const lineWidget = codeMirror.addLineWidget(
```

habrá que pasar esto como parámetro a `showFeedbackForm`.

Reply:
Aquí no supe como mandarlo .  ver si está bien como lo hice

Reply:
Parece que no está bien de la forma que lo mando (`this`), porque `this` hace referencia al componente. El problema es que `showFeedbackForm` se manda llamar justo aquí donde se inicializa `lineWidget`, así que no encuentro la forma de mandarlo. Esto también ocasiona que el botón de "Cancelar" no funcione adecuadamente.